pkg/commands/addmetadata.go

```diff
@@ -100,23 +102,33 @@ func (o *addMetadataOptions) ValidateAndParse(args []string, k KindOfAdd) error
 	}
 	inputs := strings.Split(args[0], ",")
 	for _, input := range inputs {
+		metadata := make(map[string]string)
+		var kv []string
+		//parse annotation keys and values into metadata
+		if kv = strings.Split(input, ":"); len(kv[0]) > 0 && len(kv) > 1 {
+			metadata[kv[0]] = kv[1]
+		} else if len(kv[0]) > 0 && len(kv) > 0 {
+			metadata[kv[0]] = ""
+		}
 		switch k {
 		case label:
-			valid, err := validate.IsValidLabel(input)
-			if !valid {
-				return err
+			if errs := v1validation.ValidateLabels(metadata, field.NewPath("field")); len(errs) != 0 {
+				return fmt.Errorf("invalid label format: %s", input)
```

Comment:
Noticing that the unit tests are mostly checking ValidateLabels behavior, but those functions have their own tests (one hopes).

To test just `addMetadataOptions`, could do this

```
type addMetadataOptions struct {
	metadata map[string]string
        validateLabels func(map[string]string)
        validateAnnotations func(map[string]string)
}
```

the line above would be like:
```
if errs := o.validateLabels(metadata); len(errs) != 0 {
```

and in newCmdAddLabel have something like
```
 o := addMetaDataOptions{
           make(map[string]string),
           func(x map[string]string) {v1validation.ValidateLabels(x, field.NewPath("field"))},
           func(x map[string]string) {v1validation.ValidateAnnotations(x, field.NewPath("field"))},
        }
```
and in the test just inject something fake that returns an error or not, etc.  point being that this is a test to see that the right validation is used more than anything else.